shared/studio/state/graphSettings.ts

```diff
@@ -0,0 +1,49 @@
+import {computed} from "mobx";
+import {Model, model, prop, modelAction} from "mobx-keystone";
+
+export enum graphType {
+  area = "area",
+  flame = "flame",
+}
+
+export enum graphUnit {
+  time = "time",
+  cost = "cost",
+}
+
+const userGraphTypeStorageKey = "edgedbUserGraphTypeChoice";
+
+@model("GraphSettings")
+export class GraphSettings extends Model({
+  graphType: prop<graphType>(),
+  graphUnit: prop<graphUnit>().withSetter(),
+  userUnitChoice: prop<graphUnit | null>().withSetter(),
+}) {
+  @computed
+  get isTimeGraph() {
+    return this.graphUnit === graphUnit.time;
+  }
+
+  @computed
+  get isAreaGraph() {
+    return this.graphType === graphType.area;
+  }
+
+  @modelAction
+  setGraphType(type: graphType) {
+    this.graphType = type;
+    localStorage.setItem(userGraphTypeStorageKey, JSON.stringify(type));
+  }
+}
+
+const userStoredTypeChoice = localStorage.getItem(userGraphTypeStorageKey);
+
+const graphSettings = new GraphSettings({
+  graphType: userStoredTypeChoice
+    ? JSON.parse(userStoredTypeChoice)
+    : graphType.area,
```

Comment:
I think loading from localStorage should be in a `try/catch` block and it should check the value, since the value in localStorage may be anything.
```ts
let userStoredTypeChoice = graphType.area;
try {
  const storedVal = localStorage.getItem(userGraphTypeStorageKey);
  if (storedVal) {
    userStoredTypeChoice =
      JSON.parse(storedVal) === "flame" ? graphType.flame : graphType.area;
  }
} catch {
  // ignore errors
}

const graphSettings = new GraphSettings({
  graphType: userStoredTypeChoice,
// ...
```

Reply:
hmm yes, someone else can enter the same key in local storage...